src/main/java/ac/grim/grimac/utils/collisions/HitboxData.java

```diff
@@ -264,24 +204,275 @@ public enum HitboxData {
             new HexCollisionBox(4.0, 0.0, 4.0, 12.0, 16.0, 12.0),
             BlockTags.STANDING_SIGNS.getStates().toArray(new StateType[0])),
 
-    REDSTONE_WIRE((player, item, version, data, x, y, z) ->
-            // Easier to just use no collision box
-            // Redstone wire is very complex with its collision shapes and has many de-syncs
-            NoCollisionBox.INSTANCE,
-            StateTypes.REDSTONE_WIRE),
+    SAPLING(new HexCollisionBox(2.0D, 0.0D, 2.0D, 14.0D, 12.0D, 14.0D),
+            BlockTags.SAPLINGS.getStates().toArray(new StateType[0])),
+
+    ROOTS(new HexCollisionBox(2.0D, 0.0D, 2.0D, 14.0D, 13.0D, 14.0D),
+            StateTypes.WARPED_ROOTS, StateTypes.CRIMSON_ROOTS),
+
+    BANNER(new HexCollisionBox(4.0D, 0.0D, 4.0D, 12.0D, 16.0D, 12.0D),
+            StateTypes.WHITE_BANNER, StateTypes.ORANGE_BANNER, StateTypes.MAGENTA_BANNER, StateTypes.LIGHT_BLUE_BANNER,
+            StateTypes.YELLOW_BANNER, StateTypes.LIME_BANNER, StateTypes.PINK_BANNER, StateTypes.GRAY_BANNER,
+            StateTypes.LIGHT_GRAY_BANNER, StateTypes.CYAN_BANNER, StateTypes.PURPLE_BANNER, StateTypes.BLUE_BANNER,
+            StateTypes.BROWN_BANNER, StateTypes.GREEN_BANNER, StateTypes.RED_BANNER, StateTypes.BLACK_BANNER),
+
+    WALL_BANNER((player, item, version, data, x, y, z) -> {
+        // ViaVersion replacement block
+        if (version.isOlderThan(ClientVersion.V_1_8)) {
+            return WALL_SIGN.dynamic.fetch(player, item, version, data, x, y, z);
+        }
+
+        switch (data.getFacing()) {
+            case NORTH:
+                return new HexCollisionBox(0.0, 0.0, 14.0, 16.0, 12.5, 16.0);
+            case EAST:
+                return new HexCollisionBox(0.0, 0.0, 0.0, 2.0, 12.5, 16.0);
+            case WEST:
+                return new HexCollisionBox(14.0, 0.0, 0.0, 16.0, 12.5, 16.0);
+            case SOUTH:
+                return new HexCollisionBox(0.0, 0.0, 0.0, 16.0, 12.5, 2.0);
+            default:
+                throw new RuntimeException("Impossible Banner Facing State; Something very wrong is going on");
+        }
+    }, StateTypes.WHITE_WALL_BANNER, StateTypes.ORANGE_WALL_BANNER, StateTypes.MAGENTA_WALL_BANNER,
+            StateTypes.LIGHT_BLUE_WALL_BANNER, StateTypes.YELLOW_WALL_BANNER, StateTypes.LIME_WALL_BANNER,
+            StateTypes.PINK_WALL_BANNER, StateTypes.GRAY_WALL_BANNER, StateTypes.LIGHT_GRAY_WALL_BANNER,
+            StateTypes.CYAN_WALL_BANNER, StateTypes.PURPLE_WALL_BANNER, StateTypes.BLUE_WALL_BANNER,
+            StateTypes.BROWN_WALL_BANNER, StateTypes.GREEN_WALL_BANNER, StateTypes.RED_WALL_BANNER, StateTypes.BLACK_WALL_BANNER),
+
+    BREWING_STAND((player, item, version, block, x, y, z) -> {
+        if (version.isNewerThan(ClientVersion.V_1_7_10) && version.isOlderThan(ClientVersion.V_1_9)) {
```

Comment:
`ClientVersion.V_1_8` is 1.8.0 to 1.8.9
```suggestion
        if (version.equals(ClientVersion.V_1_8)) {
```